### PR TITLE
`MOD` constraint failure on mainnet

### LINKED ISSUE #1180: MOD constraint failure on mainnet
## Ranges

```rust
2321470-2321479
```

## Issue

This range breaks the constraint

```rust
mod.set-absolute-values
```

image

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/mod/ModOperation.java [hunk 1]
```diff
@@ -127,8 +127,9 @@ private static BaseBytes getRes(OpCode op, Bytes32 arg1, Bytes32 arg2) {
   }
 
   private UInt256 absoluteValueIfSignedInst(Bytes32 arg) {
-    if (isSigned) {
-      return UInt256.valueOf(arg.toBigInteger().abs());
+    if (this.isSigned()) {
+      Bytes argBytes = Bytes.of(arg.toArray());
+      return UInt256.valueOf(argBytes.toBigInteger().abs());
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/mod/ModOperation.java:114]: This was the culprit. See screenshots here https://github.com/Consensys/linea-tracer/issues/1180#issuecomment-2351784017

Comment (letypequividelespoubelles) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mod/ModOperation.java:114]: Why not using, in the case of signed op : `return UInt256.valueOf(arg.toUnsignedBigInteger());` ?

`.toUnsignedBigInteger()` is what we use everywhere

Comment (letypequividelespoubelles) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/mod/ModOperation.java:114]: LGTM. We have to deal with https://github.com/Consensys/linea-tracer/issues/823 to have less conversions everywhere ...

File: arithmetization/src/test/java/net/consensys/linea/zktracer/LeosNewRanges.java
Comment (letypequividelespoubelles) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/LeosNewRanges.java:1]: what the point of adding so many replayTests if they were all passing ?

**Review by letypequividelespoubelles — APPROVED** [review]
LGTM, but https://github.com/Consensys/linea-tracer/issues/823. Too many conversions everywhere in this class ;(

**Comment by OlivierBBB** [comment]
I agree about the conversions. There will have to be a clean up

**Pull request merged**